tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/base/AbstractChouOrlandiOT.java

```diff
@@ -128,28 +126,22 @@ private byte[] receiveRandomOt(boolean choiceBit) {
     private Pair<byte[], byte[]> sendRandomOt() {
         // y
         BigInteger y = randNum.nextBigInteger(getDhModulus());
-        // S
-        T S = this.getGenerator().exponentiation(y);
-        network.send(otherId, S.toByteArray());
-        // T = G(S)
-        T T = hashToFieldElement(S, "FRESCO|ChouOrlandi|OT");
+        // A
+        T A = this.getGenerator().exponentiation(y);
+        network.send(otherId, A.toByteArray());
         byte[] rBytes = network.receive(otherId);
-        // R
-        T R = decodeElement(rBytes);
+        // U
+        T U = decodeElement(rBytes);
 
         byte[] k0Hash, k1Hash;
-        hashDigest.update(S.toByteArray());
-        hashDigest.update(R.toByteArray());
-        // R^y*T^(-jy) with j = 0:
-        // R^y * T^(-0y) == R^y
-        hashDigest.update(R.exponentiation(y).toByteArray());
+        hashDigest.update(A.toByteArray());
+        // (U * B^(0))^y == U^y
+        hashDigest.update(U.exponentiation(y).toByteArray());
         k0Hash = hashDigest.digest();
 
-        hashDigest.update(S.toByteArray());
-        hashDigest.update(R.toByteArray());
-        // R^y*T^(-jy) with j = 1:
-        // R^y * T^(-1y) == (R + (-T)) * y
-        hashDigest.update((R.groupOp(T.inverse())).exponentiation(y).toByteArray());
+        hashDigest.update(A.toByteArray());
+        // (U * B^(-1))^y
+        hashDigest.update((U.groupOp(A.inverse())).exponentiation(y).toByteArray());
```

Comment:
In the following SHAKE is used.
Why use both SHA256 and SHAKE? Would it not make sense to use the same hash function?

Reply:
I'm not sure I am following, the later `hashToFieldElement` which uses SHAKE is never called, and the highlighted code only uses SHA-256 for hashing?

Reply:
If SHAKE is never used then the code should be removed. Currently there is code using both kinds, but if the SHAKE code is now dead it should just be removed

Reply:
Resolved by 600750daa

Reply:
Solved by 20615343b